core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -44,8 +44,9 @@
  * can construct different queries based on the type provided.
  */
 public class MatchQueryBuilder extends AbstractQueryBuilder<MatchQueryBuilder> {
-    public static final ParseField MATCH_PHRASE_FIELD = new ParseField("match_phrase", "text_phrase");
-    public static final ParseField MATCH_PHRASE_PREFIX_FIELD = new ParseField("match_phrase_prefix", "text_phrase_prefix");
+    public static final ParseField MATCH_PHRASE_FIELD = new ParseField("match_phrase", "text_phrase").withAllDeprecated("match");
+    public static final ParseField MATCH_PHRASE_PREFIX_FIELD = new ParseField("match_phrase_prefix", "text_phrase_prefix")
+            .withAllDeprecated("match");
     public static final ParseField SLOP_FIELD = new ParseField("slop", "phrase_slop");
     public static final ParseField ZERO_TERMS_QUERY_FIELD = new ParseField("zero_terms_query");
     public static final ParseField CUTOFF_FREQUENCY_FIELD = new ParseField("cutoff_frequency");
```

Comment:
I think we should deprecate the type parse field too and have some test for that?